package/MDAnalysis/core/Selection.py

```diff
@@ -178,6 +178,12 @@ def __init__(self, parser, tokens):
         pass
 
     def apply(self, group):
+        # Check whether group is identical to the one stored
+        # in the corresponding universe, in which case this
+        # is returned directly. This works since the Universe.atoms
+        # are unique by construction.
+        if group.universe and group is group.universe.atoms:
```

Comment:
Yeah moving forward **everything** will have a Universe

@wouterboomsma I'd just change it to `if group is group.universe.atoms`